### DISCUSSION TITLE
with v99 you might need to manually update the provider field in directus_users if using external auth

### DISCUSSION THREAD

**Discussion by wc-matteo** [directus/directus#9049] [Archive]
I'm using `auth0`

**Before** v99 I had `default` as the value of `provider` in `directus_users`

**On** v99 I had to update (through the database) `default` to `auth0` (matching `AUTH_PROVIDERS` env var) to be able to log in. With `default` I was getting:

```json
{"errors":[{"message":"Invalid user credentials.","extensions":{"code":"INVALID_CREDENTIALS"}}]}
```

**Comment by wc-matteo** [comment]
there's seems to be something wrong still. Sometimes when I try to login I get: `INVALID_CREDENTIALS`. Refresh. It says I'm logged in and if I want to continue. Continue and it's fine

**Reply by azrikahar** [reply]
cc @aidenfoxx

**Reply by aidenfoxx** [reply]
@wc-matteo You're using OpenID I assume? That is correct that if you want to continue using existing OpenID users you have to manually update the `provider` field in the database to the correct provider name. Sorry about that!!

As for the incorrect redirect, I believe that is cause by: https://github.com/directus/directus/issues/9053

You can test if the redirect issue is caused by the above bug by setting a query parameter on the URL before clicking an SSO link e.g. `/admin/login?test`. If you're able to log in correctly then the issue was caused by the bug.

**Reply by wc-matteo** [reply]
@aidenfoxx thanks for the reply

> You're using OpenID I assume

No, oauth2

> #9053

Yep, I saw that `?continue` at the start of the url and didn't know if it was intended or not


Let's see what happens

**Reply by wc-matteo** [reply]
> You can test if the redirect issue is caused by the above bug by setting a query parameter on the URL before clicking an SSO link e.g. /admin/login?test. If you're able to log in correctly then the issue was caused by the bug.

yep that was the issue
